pkg/sink/kafka/config.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"crypto/x509"
+	"strings"
+
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/pkg/security"
+	"github.com/pkg/errors"
+	"go.uber.org/zap/zapcore"
+)
+
+var (
+	ProtocolDefault  = "plaintext"
+	ProtocolSSL      = "ssl"
+	ProtocolSASL     = "sasl_plaintext"
+	ProtocolSASL_SSL = "sasl_ssl"
+)
+
+func NewConfig(options *Options) *kafka.ConfigMap {
+	config := &kafka.ConfigMap{
+		"bootstrap.servers": strings.Join(options.BrokerEndpoints, ","),
+		"log_level":         getLogLevel(),
+	}
+	if options.EnableTLS {
+		_ = config.SetKey("security.protocol", ProtocolSSL)
+		if options.Credential != nil && options.Credential.IsTLSEnabled() {
+			_ = config.SetKey("ssl.ca.location", options.Credential.CAPath)
+			_ = config.SetKey("ssl.certificate.location", options.Credential.CertPath)
+			_ = config.SetKey("ssl.key.location", options.Credential.KeyPath)
+			addVerifyPeerCertificate(config, options.Credential.CertAllowedCN)
+
+		}
+		_ = config.SetKey("enable.ssl.certificate.verification", !options.InsecureSkipVerify)
+	}
+
+	completeSASLConfig(config, options)
+
+	compression := strings.ToLower(strings.TrimSpace(options.Compression))
+	config.SetKey("compression.codec", compression)
+	if compression != "none" {
+		log.Info("Kafka producer uses " + compression + " compression algorithm")
```

Comment:
no need to print log here, but print log at the last line, print all important fields.

Reply:
Print log for kafka compression integration test

Reply:
I cannot understand why a integration test depend on a log, this is not good

Reply:
The test gets the compression algorithm based on logs.
https://github.com/pingcap/ticdc/blob/eee5de253e870d4a15ddcc200a93904d015c0a0a/tests/integration_tests/kafka_compression/run.sh#L21